internal/nodes/avi_model_evh_nodes.go

```diff
@@ -2092,9 +2126,11 @@ func DeleteDedicatedEvhVSNode(vsNode *AviEvhVsNode, key string, hostsToRemove []
 func manipulateEvhNodeForSSL(key string, vsNode *AviEvhVsNode, evhNode *AviEvhVsNode) {
 	oldSSLProfile := vsNode.GetSSLProfileRef()
 	newSSLProfile := evhNode.GetSSLProfileRef()
-	if oldSSLProfile != "" && oldSSLProfile != newSSLProfile {
+	if oldSSLProfile != nil &&
+		*oldSSLProfile != "" &&
+		oldSSLProfile != newSSLProfile {
```

Comment:
are we missing a nil check of newSSLProfile here?

Reply:
Nil check not required as newSSLProfile will not be nil. Also, I have modified the condition from
```
if oldSSLProfile != nil &&
		*oldSSLProfile != "" &&
		oldSSLProfile != newSSLProfile {
		utils.AviLog.Warnf("key: %s msg: overwriting old ssl profile %s with new ssl profile %s", key, oldSSLProfile, newSSLProfile)
}
```
to 
```
if oldSSLProfile != nil &&
		*oldSSLProfile != "" &&
		*oldSSLProfile != *newSSLProfile {
		utils.AviLog.Warnf("key: %s msg: overwriting old ssl profile %s with new ssl profile %s", key, oldSSLProfile, newSSLProfile)
}
```
as we would end up comparing the pointers instead of values here `oldSSLProfile != newSSLProfile {`